### PR TITLE
[HarfBuzz] Update Port to 2.8.1

### PR DESCRIPTION
**Pull request by kirawi**
**Describe the pull request**
Updates HarfBuzz to 2.8.1

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No change

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by kirawi** [comment]
`REGRESSION: paraview:x86-windows. If expected, add paraview:x86-windows=fail to .\scripts\ci.baseline.txt.` Seems unrelated to the PR.

File: ports/harfbuzz/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "harfbuzz",
-  "version-string": "2.7.4",
+  "version-string": "2.8.0",
   "port-version": 1,
```
Comment (JonLiu1993) [inline ports/harfbuzz/vcpkg.json:4]: @kirawi ,please delete  "port-version",

File: ports/harfbuzz/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "harfbuzz",
-  "version-string": "2.7.4",
+  "version-string": "2.8.0",
```
Comment (JonLiu1993) [inline ports/harfbuzz/vcpkg.json:3]: please change "version-string" to "version"

**Review by JonLiu1993 — COMMENTED** [review]
@kirawi,please Keep the format of patch2 correct

**Comment by JonLiu1993** [comment]
@kirawi ,please take a look : 
````
CMake Error at D:/downloads/tools/cmake-3.19.2-windows/cmake-3.19.2-win32-x86/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find Gettext (missing: GETTEXT_MSGMERGE_EXECUTABLE
  GETTEXT_MSGFMT_EXECUTABLE)
````

**Comment by JonLiu1993** [comment]
The another failures caused by popsift will be fixed by #17277.
````
CMake Error at ports/popsift/portfile.cmake:9 (include):
  include could not find load file:

    D:/installed/x64-windows/share/vcpkg_find_cuda/vcpkg_find_cuda.cmake
````

**Comment by kirawi** [comment]
> @kirawi,please Keep the format of patch2 correct

Sorry, could you elaborate on this as well as the CMake error you commented? I'm afraid I don't quite understand.

**Comment by JonLiu1993** [comment]
@kirawi ,Sometimes the error in CI may not be the problem of the port you modified, it may be caused by the port you rely on or other port errors on CI. At present, all CI problems have been fixed, and your pr is ready to merge.

**Comment by JonLiu1993** [comment]
@kirawi ，please modify the content I marked, and then I can mark review

**Comment by kirawi** [comment]
All checks have passed.

**Comment by strega-nil-ms** [comment]
@kirawi can you merge with latest?

**Comment by kirawi** [comment]
> @kirawi can you merge with latest?

Should be done now.

**Comment by kirawi** [comment]
The failing checks seemed to be caused by other packages.

**Comment by kirawi** [comment]
CI froze at these points:
- x64_windows: kf5config[core]
- x64_windows_static_md: kf5guiaddons[core]
- x86_windows: openmvg[core,software[

**Comment by kirawi** [comment]
CI froze at these points since at least 2 hours ago:

- x64_windows: kf5config[core]
- x64_windows_static_md: kf5guiaddons[core]
- x86_windows: openmvg[core,software[

**Comment by kirawi** [comment]
Looks like x64_windows froze on marble[core], and x64_windows_static_md failed to build `opencv4` and `kf5itemviews`. CI is not an easy beast to work with ^_^

**Comment by kirawi** [comment]
x86_windows failed on an unrelated port.

**Comment by PhoebeHui** [comment]
@dan-shaw, could you help merge this PR?

File: versions/h-/harfbuzz.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "3f1e0749ade6fb27bbb6103e1faab23b2719dcda",
-      "version": "2.7.4",
-      "port-version": 2
+      "git-tree": "14b26ebe0122cd5546c795d988ac677c31d3afa8",
+      "version": "2.8.0",
+      "port-version": 0
```
Comment (dan-shaw) [inline versions/h-/harfbuzz.json:6]: We should add a new entry to the version list instead, so users can reference a previous version.

**Review by PhoebeHui — APPROVED** [review]

**Comment by kirawi** [comment]
I'll go ahead and bump the version to 2.8.1

File: versions/h-/harfbuzz.json [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "32e5438e1f8c76646b657f14fbe0dac7646d80f6",
+      "version": "2.8.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "14b26ebe0122cd5546c795d988ac677c31d3afa8",
+      "version": "2.8.0",
+      "port-version": 0
+    },
```
Comment (JonLiu1993) [inline versions/h-/harfbuzz.json:12]: @kirawi ,When we update the version, we only need to add "version" once, please delete this. Thanks

**Pull request merged**